### ISSUE TITLE
Define Work Schedule with Weekdays and Weeknights

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#1200]
**Is your feature request related to a problem? Please describe.**
- Currently, each settlement can have one type of work shift or work schedule. It stays the same on every sol and doesn't distinguish between weekday or weekend. 
 
**Describe the rationale**
- Define # of work days for each settlement. 
- May set to 5 or 6 days for each week, thus having 1 or 2 day off .

**Describe the goal(s) you have in mind**
- For 4-person settlement, a work day would probably be 500 millisols per shift and 6 days or work per week.
- However, for a 36-person settlement, each settler could afford a 333-millisol shift and 5 days per week schedule.

**Describe the solution you desire to see**
- Allow define which weekday(s) as the *on* day and *off* day.

**Specifications**
- post 3.7.0 build 9180

**Additional context**
- Eventually, allow various types of work schedule to coexist (say, for each type of department/divisions) in a settlement. e.g. In some developed country on Earth, a nurse may have 4 days of work, followed by 3 days of leaves.

**Comment by bevans2000** [comment]
This can be done by extending the ShiftSpec design. But how do you want it to work?

- A single Shift has different times for different days? So the same Persons are always assigned to a particular Shift?
- Do weekends really exist on a Space Station? Every day is a work day?

**Comment by mokun** [comment]
> A single Shift has different times for different days? 

No. We'll stick with a single shift has only one start time and one end time.

> Do weekends really exist on a Space Station? Every day is a work day?

Yes. I used to work in military base oversea and it causes low moral if a person has to work 7 days straight.

So, I'd like to see the ability to have a day off after working for x numbers of days for each person.

Say for a 12-person settlement. 3 of them would have a Friday off. 3 having Saturday off. 3 having Sunday off. 3 having Monday off.

Eventually, each career field or each division may choose its own shift schedule.

But for now, it makes sense to allow each settler to have one or two *off* day .

**Comment by bevans2000** [comment]
Ok, easy. So for each ShiftSpec we define the maximum days worked consecutively. Then when assisting the Shift we randomly spread the days off so everyone gets a day off during the week. We already have the concept of On Leave when a Person changes Shift.
